### DISCUSSION TITLE
Appreciate the graphs!

### DISCUSSION THREAD

**Discussion by jflyons** [mcguirepr89/BirdNET-Pi#779] [General]
Enjoying my BirdNET-Pi all the more with the on-demand species detection graphs!

**Comment by ehpersonal38** [comment]
Hey - you're welcome! Glad to see that people enjoy the little updates I push. 

In my area the Eastern Bluebirds just started nesting this past month, and you can definitely tell!
![chrome_nvKK3tvUCx](https://user-images.githubusercontent.com/103586016/224197770-699b8fa3-a368-459f-88a2-bd116108e48d.png)

**Reply by blogdemoi** [reply]
It is a very nice update, I like it.

**Reply by Fish31520** [reply]
@ehpersonal38 Nice update, very good job, as usual !

**Comment by lloydbayley** [comment]
God Damn, @ehpersonal38!!!!!! You snuck that one in didn't you?!
I only just updated and now BAM! 15 new updates. You're doing well! 
I need to put the Github sensors back into my Home Assistant config so I know when you've done more good work! :)

**Comment by becausebirds** [comment]
I'm not seeing any updates in my system. Is there something I should do to get the latest updates to show in system controls? I've never experienced this before.

**Reply by ehpersonal38** [reply]
I believe this is happening because in the past week I've "rebased" the main branch of this repo a few times, which messes with the update count. But it's just a cosmetic issue, so even though the update count does not show up on your system, clicking the "Update" button will successfully update your installation.

I issued a fix for this so it won't happen again in the future - 7e825d600944e404efcd69bd65cdcad1bafa20af.

**Reply by becausebirds** [reply]
I've tried the update multiple times and it seems to fail each time. This is a portion of the output 
<img width="1100" alt="Screenshot 2023-03-11 at 2 25 04 PM" src="https://user-images.githubusercontent.com/110354170/224510095-df1d78d9-d65c-44e3-9c00-3786450fca9b.png">

**Reply by ehpersonal38** [reply]
Try removing `.git/ORIG_HEAD`:
`rm ~/BirdNET-Pi/.git/ORIG_HEAD`

Then retry updating. If that doesn't work then remove `HEAD` as well:
`rm ~/BirdNET-Pi/.git/HEAD`

**Reply by becausebirds** [reply]
That definitely got rid of the errors that were showing, I had to run both commands, but I'm not seeing any of the new graph features I can't tell what version I'm on. Thank you for your help, as always!

**Reply by ehpersonal38** [reply]
Can you run the update again and then copy/paste or screenshot the log output from that? And you're welcome!

**Reply by becausebirds** [reply]
Here is most of it
<img width="1030" alt="Screenshot 2023-03-11 at 5 52 30 PM" src="https://user-images.githubusercontent.com/110354170/224516738-24ea9ad8-8c3f-4897-9d21-185d07dd8f92.png">

**Reply by ehpersonal38** [reply]
See if these commands work:
```
cd ~/BirdNET-Pi
git remote rm origin
git remote add origin https://github.com/mcguirepr89/BirdNET-Pi
git fetch --all
```

But I suspect they won't and you'll still see a fatal message, in which case try:
`git init && git fetch --all`

**Reply by becausebirds** [reply]
I ran the commands and I see something totally different (caddy I think) when accessing birdnet.local. 
<img width="1093" alt="Screenshot 2023-03-11 at 8 09 33 PM" src="https://user-images.githubusercontent.com/110354170/224520430-012bd229-c750-4e9a-b315-32e410b30666.png">

<img width="1281" alt="Screenshot 2023-03-11 at 8 10 24 PM" src="https://user-images.githubusercontent.com/110354170/224520437-9d4e69bd-13a2-4952-9188-3eeb40c9f0b8.png">

**Reply by ehpersonal38** [reply]
Ouch, I'm sorry about that. I think I broke your installation. At this point I'd recommend backing up your `scripts/birds.db` and `birdnet.conf`, and any other custom scripts or stuff you have, then following the uninstall instructions on this repo's Github page. Then you should be able to reinstall, replace your backups, restart your services, and hopefully everything will be alright.

**Reply by becausebirds** [reply]
Thank you! I reinstalled and restored all the critical files and it seems to be going great now. <3

**Comment by lloydbayley** [comment]
I won't miss any more updates now. I've written a small script & modified Phill Price's mushroom code chunk to include a red badge-dot when there's an update.

![5926D508-4B4D-45AF-B9C8-69D41D16ABF4](https://user-images.githubusercontent.com/21309308/224541212-ebb97fe4-0ba2-4515-8a70-363588e9b817.jpeg)

**Reply by ehpersonal38** [reply]
This is cool!! I've just pushed a few minor updates, so you can give it a test. Also I finally got around to colorizing the output of Update, so it shows exactly which files were modified and by how much. I think it looks pretty sweet.

**Reply by lloydbayley** [reply]
It VERKS!!!
<img width="386" alt="image" src="https://user-images.githubusercontent.com/21309308/224645308-2bd1edd2-b901-4bf7-a224-2bd60c6ab656.png">
(I did forget an absolute path in the script, so yes, it was a good test!)
